src/main/java/com/google/cloud/spanner/jdbc/JdbcDatabaseMetaData.java

```diff
@@ -1495,16 +1496,68 @@ public boolean autoCommitFailureClosesAllResultSets() throws SQLException {
     return false;
   }
 
-  @Override
-  public ResultSet getClientInfoProperties() throws SQLException {
+  /**
+   * The max length for client info values is 63 to make them fit in Cloud Spanner session labels.
+   */
+  static final int MAX_CLIENT_INFO_VALUE_LENGTH = 63;
+
+  static Properties getDefaultClientInfoProperties() throws SQLException {
+    Properties info = new Properties();
+    try (ResultSet rs = getDefaultClientInfo()) {
+      while (rs.next()) {
+        info.put(rs.getString("NAME"), rs.getString("DEFAULT_VALUE"));
+      }
+    }
+    return info;
+  }
+
+  private static ResultSet getDefaultClientInfo() throws SQLException {
     return JdbcResultSet.of(
         ResultSets.forRows(
             Type.struct(
                 StructField.of("NAME", Type.string()),
-                StructField.of("MAX_LEN", Type.string()),
+                StructField.of("MAX_LEN", Type.int64()),
                 StructField.of("DEFAULT_VALUE", Type.string()),
                 StructField.of("DESCRIPTION", Type.string())),
-            Collections.<Struct>emptyList()));
+            Arrays.asList(
```

Comment:
These default `ClientInfo` properties are described here: https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html#setClientInfo-java.lang.String-java.lang.String-

The names are case insensitive.